### PR TITLE
#44 [FrontApp] Add user authorization for doing a task

### PR DESCRIPTION
**Pull request by Lomey**

### CODE REVIEW

File: app/logic/tasks.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module Tasks
+  extend self
+
```
Comment (sveredyuk) [inline app/logic/tasks.rb]: I think we can yet keep all logic in the `Competition` module as this logic 100% relates to competition.

File: app/logic/tasks/assessment_calculator.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Tasks
+  class AssessmentCalculator
```
Comment (sveredyuk) [inline app/logic/tasks/assessment_calculator.rb]: spec this

File: spec/logic/tasks/user_task_policy_participant_spec.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    end
+  end
+
+  context 'when a task is started but with missing dependent task' do
+    it { expect(policy).to be true }
+  end
+
+  context 'when a dependent task assessed enough' do
+    let(:dependent_task) do
+      create(:task,
```
Comment (sveredyuk) [inline spec/logic/tasks/user_task_policy_participant_spec.rb]: better 3-lines formatting:
```ruby
create(
 -,
 -,
 something: {
    -,
    -
  }
)
```

**Review by sveredyuk — COMMENTED** [review]
Good job

File: spec/logic/tasks/user_task_policy_participant_spec.rb (hunk 3 above)
Comment (Lomey) [reply spec/logic/tasks/user_task_policy_participant_spec.rb]: Done

File: app/logic/tasks.rb (hunk 1 above)
Comment (Lomey) [reply app/logic/tasks.rb]: Done

File: app/logic/tasks/assessment_calculator.rb (hunk 2 above)
Comment (Lomey) [reply app/logic/tasks/assessment_calculator.rb]: Done

**Review by sveredyuk — APPROVED** [review]
Well done 👍

**Pull request merged**